### DISCUSSION TITLE
WebView.EvaluateJavaScriptAsync task never completes.

### DISCUSSION THREAD

**Discussion by Ghostbird** [dotnet/maui#21448] [Q&A] [answered]
Very often when I call `EvaluateJavaScriptAsync` on my `WebView`, the task never completes. Code after the `await` of the task is never executed, and in the attached chrome remote debugger I can see that none of the results of the JavaScript I tried to run are present in the WebView state either. It's as if the JavaScript has never been run. Although it seems that this function will silently swallow all JavaScript exceptions, so I can't guarantee that's what has happened. I build the JavaScript string in C#, and if I pause the debugger before the call to `EvaluateJavaScriptAsync`, and instead manually paste the built string in the attached chrome remote debugger console, it runs without error.

Does anyone have an idea what I'm doing wrong?

**Comment by mycroes** [comment]
Did you fix this issue? Having the exact same problem...

**Reply by Ghostbird** [reply]
We no longer have this issue, but I don't recall what I did to fix it. I think it was an error in the javascript that was completely silent.

**Reply by mycroes** [reply, marked as answer]
Found the problem. There's a bug that doesn't execute multiline JS. Combining all into a single line of JavaScript (with semicolons, braces, etc.) does work.

**Reply by Ghostbird** [reply]
Oh, that might have been our issue too. I do recall that I rewrote some stuff to use more complex setter logic on the JavaScript side to try to solve this. As a side effect, that might have reduced my statement called in `EvaluateJavascriptAsync` to a single line.

**Reply by hammondreg** [reply]
> Found the problem. There's a bug that doesn't execute multiline JS. Combining all into a single line of JavaScript (with semicolons, braces, etc.) does work.

Thank you Michael, you saved my sorry ass and a project that's seventeen years in the making from going down the drain

**Reply by mycroes** [reply]
> > Found the problem. There's a bug that doesn't execute multiline JS. Combining all into a single line of JavaScript (with semicolons, braces, etc.) does work.
> 
> Thank you Michael, you saved my sorry ass and a project that's seventeen years in the making from going down the drain

You're welcome, happy to have helped!

**Comment by hammondreg** [comment]
You did indeed help.

ChatGPT messed me around for over a day without spotting the fatal problem ... so you're smarter than chatGPT!!

________________________________
